geometric_kernels/kernels/product.py

```diff
@@ -0,0 +1,85 @@
+"""
+Product of kernels
+"""
+from typing import List, Mapping, Tuple
+
+import lab as B
+
+from geometric_kernels.kernels import BaseGeometricKernel
+from geometric_kernels.spaces import Space
+
+
+class ProductGeometricKernel(BaseGeometricKernel):
+    """
+    Basic implementation of product kernels.
+    TODO: Support combined RFF/KL expansion methods
+
+    Parameters
+    ----------
+    kernels : BaseGeometricKernel
+        kernels to compute the product for
+    dimension_indices : List[B.Numeric], optional
+        List of indices the correspond to the indices of the input that should be fed to each kernel.
+        If None, assume the each kernel takes kernel.space.dimension inputs, and that the input will
+        be a stack of this size, by default None
```

Comment:
Please move the docstrings to the `__init__`. I would also remove the type information as this is redundant.